src/test/javascript/spec/component/exam/test-run/create-test-run-modal.component.spec.ts

```diff
@@ -1,51 +1,88 @@
     const exam = { id: 1, course, started: true, startDate: moment(), endDate: moment().add(20, 'seconds'), exerciseGroups: [exerciseGroup1] } as Exam;
     const exerciseGroup2 = { id: 2 } as ExerciseGroup;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [ArtemisTestModule],
+            imports: [FormsModule, ReactiveFormsModule],
             declarations: [CreateTestRunModalComponent],
-            providers: [],
-        })
-            .overrideTemplate(CreateTestRunModalComponent, '')
-            .compileComponents();
+            providers: [NgbModal, NgbActiveModal, { provide: ArtemisDurationFromSecondsPipe, useClass: ArtemisDurationFromSecondsPipe }],
+        }).compileComponents();
 
         fixture = TestBed.createComponent(CreateTestRunModalComponent);
         comp = fixture.componentInstance;
     });
 
+    afterEach(() => {
+        sinon.restore();
+    });
+
     describe('OnInit', () => {
         it('should initialise the working time form ', fakeAsync(() => {
             comp.exam = exam;
             // WHEN
             comp.ngOnInit();
             // THEN
-            expect(!!comp.workingTimeForm).toBeTruthy();
+            expect(!!comp.workingTimeForm).to.be.ok;
         }));
     });
 
+    describe('creating test run workflow', () => {
+        it('should create a new test run and close the modal', () => {
+            const activeModal = TestBed.inject(NgbActiveModal);
+            const closeStub = sinon.stub(activeModal, 'close');
+            comp.exam = exam;
+            fixture.detectChanges();
+            comp.workingTimeForm.controls['minutes'].setValue(30);
+            comp.workingTimeForm.controls['seconds'].setValue(0);
+            const exerciseRow = fixture.debugElement.query(By.css('#exercise-1')).nativeElement;
+            expect(exerciseRow).to.be.ok;
+            exerciseRow.click();
+            fixture.detectChanges();
+            expect(comp.testRunConfiguration[1]).to.deep.equal(exercise);
+            expect(comp.exam.exerciseGroups!.length).to.equal(1);
+            expect(comp.testRunConfigured).to.equal(true);
+            const createTestRunButton = fixture.debugElement.query(By.css('#createTestRunButton')).nativeElement;
+            createTestRunButton.click();
+            expect(closeStub).to.have.been.called;
+            const testRun = closeStub.getCall(0).args[0] as StudentExam;
+            expect(testRun).to.be.ok;
+            expect(testRun.exam).to.equal(exam);
+            expect(testRun.exercises).to.contain(exercise);
+            expect(testRun.workingTime).to.equal(1800);
+        });
+    });
+
     describe('Ignore Exercise groups', () => {
         it('should ignore exercise groups with no exercises', function () {
             comp.exam = exam;
             comp.exam.exerciseGroups = [exerciseGroup1, exerciseGroup2];
             fixture.detectChanges();
-            expect(comp.exam.exerciseGroups!.length).toBe(1);
```

Comment:
This test doesn't do anything anymore if I see this correctly? (No except statement.)

Reply:
good catch added it back in :) done in b69401a5aabd8b074c285e88405213360de6465b